src/Scientist/Internals/Experiment.cs

```diff
@@ -18,6 +18,7 @@ internal class Experiment<T> : IExperiment<T>, IExperimentAsync<T>
         Func<Task> _beforeRun;
         Func<Task<bool>> _runIf = _alwaysRun;
         readonly List<Func<T, T, Task<bool>>> _ignores = new List<Func<T, T, Task<bool>>>();
+        readonly Dictionary<string, object> _contexts = new Dictionary<string, object>();
```

Comment:
Upon thinking about it, would this be better as a `Dictionary<string, dynamic>`? Hopefully would mean less casting when publishing.